### PR TITLE
feat: support disableCssExtract

### PR DESCRIPTION
**Pull request by targeral**
# PR Details

support output.disableCssExtract

## Types of changes

- [x] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My code follows the code style of this project.
- [x] I have updated changeset
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: packages/cli/webpack/src/utils/createCSSRule.ts [hunk 1]
```diff
@@ -44,6 +48,10 @@ export const createCSSRule = (
       chain.output.get('publicPath') === './' ? { publicPath: '../../' } : {},
     )
     .end()
+    .when(disableCssExtract && isProd(), c => {
+      c.uses.delete('mini-css-extract');
```
Comment (chenjiahan) [inline packages/cli/webpack/src/utils/createCSSRule.ts:52]: 这里 delete 的逻辑稍微有点不直观，是不是可以在上面 `use('mini-css-extract')` 的时候直接加条件判断，而不是 use 之后再 delete

File: packages/cli/webpack/src/config/base.ts [hunk 2]
```diff
@@ -472,14 +472,15 @@ class BaseWebpackConfig {
...
         .use(WebpackBar, [{ name: this.chain.get('name') }]);
-
-    this.chain.plugin('mini-css-extract').use(MiniCssExtractPlugin, [
-      {
-        filename: this.cssChunkname,
-        chunkFilename: this.cssChunkname,
-        ignoreOrder: true,
-      },
-    ]);
+    if (!(this.options.output?.disableCssExtract && isProd())) {
```
Comment (chenjiahan) [inline packages/cli/webpack/src/config/base.ts:475]: 「只在生产环境启用」的收益是什么嘞，是不是可以考虑让 dev 和 build 保持一致

Comment (chenjiahan) [reply packages/cli/webpack/src/config/base.ts:475]: 我去调研一下其他框架哈

Comment (chenjiahan) [reply packages/cli/webpack/src/config/base.ts:475]: 看了下，应该只在生产环境启用更合适一些。属于 webpack 官方推荐用法：image

**Review by chenjiahan — APPROVED** [review]

File: packages/cli/webpack/src/config/base.ts (hunk 2 above)
Comment (chenjiahan) [inline packages/cli/webpack/src/config/base.ts:475]: 应该是 `!this.options.output?.disableCssExtract && isProd()`

File: packages/cli/webpack/src/utils/createCSSRule.ts [hunk 3]
```diff
@@ -32,18 +33,28 @@ export const createCSSRule = (
...
   loaders
     .oneOf(name)
     .test(test)
-    .use('mini-css-extract')
-    .loader(require('mini-css-extract-plugin').loader)
-    .options(
-      chain.output.get('publicPath') === './' ? { publicPath: '../../' } : {},
-    )
-    .end()
+    .when(!(disableCssExtract && isProd()), c => {
```
Comment (chenjiahan) [inline packages/cli/webpack/src/utils/createCSSRule.ts:45]: 应该是 `!disableCssExtract && isProd()`

File: packages/cli/webpack/src/utils/createCSSRule.ts [hunk 4]
```diff
@@ -32,18 +33,28 @@ export const createCSSRule = (
...
+      c.use('mini-css-extract')
+        .loader(require('mini-css-extract-plugin').loader)
+        .options(
+          chain.output.get('publicPath') === './'
+            ? { publicPath: '../../' }
+            : {},
+        )
+        .end();
+    })
+    .when(disableCssExtract && isProd(), c => {
```
Comment (chenjiahan) [inline packages/cli/webpack/src/utils/createCSSRule.ts:55]: 应该是 `disableCssExtract || !isProd()`

File: packages/cli/webpack/src/utils/createCSSRule.ts (hunk 3 above)
Comment (chenjiahan) [reply packages/cli/webpack/src/utils/createCSSRule.ts:45]: 建议提取一个变量，这样本文件里的两处代码就不用写重复的逻辑了

```js
const isExtractCSS = !disableCssExtract && isProd();
```

**Review by chenjiahan — CHANGES_REQUESTED** [review]

File: packages/cli/webpack/src/config/base.ts (hunk 2 above)
Comment (targeral) [reply packages/cli/webpack/src/config/base.ts:475]: 不是吧，应该是 `!this.options.output?.disableCssExtract || !isProd()`

File: packages/cli/webpack/src/utils/createCSSRule.ts (hunk 3 above)
Comment (targeral) [reply packages/cli/webpack/src/utils/createCSSRule.ts:45]: OK

File: packages/cli/webpack/src/config/base.ts (hunk 2 above)
Comment (chenjiahan) [reply packages/cli/webpack/src/config/base.ts:475]: MiniCssExtractPlugin 不是只在生产环境下启用嘛，为啥是 `|| !isProd()`

Comment (targeral) [reply packages/cli/webpack/src/config/base.ts:475]: 🙆‍♂️

**Review by chenjiahan — APPROVED** [review]

**Pull request merged**